feat: add textfield doc (#40)
wonderflow-bv/design: pages/develop/components/textfield.mdx

import { MdxLayout } from "@/components/layouts/mdx";
import { LiveArea } from "@/components/live-area";
import { Props } from "@/components/props";
import {
  Textfield,
  Snackbar,
  Button,
  Stack,
} from "@wonderflow/react-components";

export default ({ children }) => (
  <MdxLayout title="Textfield" description="Input areas for data entry">
    {children}
  </MdxLayout>
);

```js
import { Textfield } from "@wonderflow/react-components";
```

> With the `Textfield` component is a dynamic input field that reacts to entered data, providing a basic content format validation feedback and custom icons.

Reviewer: ```suggestion
The `Textfield` component is a dynamic input field that reacts to entered data, providing a basic content format validation feedback and custom icons.
```

## Usage

> By default the Textfield component just renders a `input` element without any type declared. To define a field of type `text` you need then to specify the `type="text"` attribute.

Reviewer: ```suggestion
By default, the Textfield component just renders an `input` element without any type declared. To define a field of type `text` you need then to specify the `type="text"` attribute.
```

<LiveArea>
  <Textfield
    type="text"
    id="filled"
    label="Filled"
    defaultValue="Prefilled value"
  />
</LiveArea>

```jsx
<Textfield
  type="text"
  id="demo1"
  label="Label"
  defaultValue="Prefilled value"
/>
```

> Since Wanda uses HTML native elements where possibile there are many different input types you can use, most of them are handled by the platform so you don't need to care about positioning, scroll, overflow, OS themes, high contrast etc. Here a list of the available types:

Reviewer: ```suggestion
Since Wanda uses HTML native elements where possible there are many different input types you can use, most of them are handled by the platform so you don't need to care about positioning, scroll, overflow, OS themes, high contrast, etc. Here is a list of the available types:
```

<Textfield type="text" label="Empty disbled" placeholder="Placeholder" disabled />
<Textfield type="text" label="Filled disabled" defaultValue="Prefilled value" disabled />
<Textfield type="number" label="Type number" defaultValue={100} />
<Textfield type="search" label="Type search" />
<Textfield type="date" label="Type date" />
<Textfield type="time" label="Type time" />
<Textfield type="month" label="Type month" />
<Textfield type="week" label="Type week" />
<Textfield type="datetime-local" label="Type datetime-local" />
```

## Sizes and icons

Like other components, the size of thr `Textfield` can be controlled with the `size` property. If needed you can also add an icon to convey meaning.

<LiveArea>
  <Stack verticalAlign="end" direction="row" columnGap={8}>
    <Textfield
      type="date"
      size="big"
      icon="calendar"
      type="text"
      label="Label"
    />
    <Button size="big">Confirm</Button>
  </Stack>
</LiveArea>

```jsx
<Stack verticalAlign="end" direction="row" columnGap={8}>
  <Textfield type="date" size="big" icon="calendar" type="text" label="Label" />
  <Button size="big">Confirm</Button>
</Stack>
```

## Data validation

> The input `validation` property can be used to provide a visual feedback about the validation of the entered data. This feedback is meant to be used only when the data doesn't match a specific format required by field.

Reviewer: ```suggestion
The input `validation` property can be used to provide visual feedback about the validation of entered data. This feedback is meant to be used only when the data doesn't match a specific format required by the field.
```
